contrib/setup-btor2tools.sh

```diff
@@ -9,15 +9,8 @@ git clone --depth 1 https://github.com/Boolector/btor2tools.git ${BTOR2TOOLS_DIR
 cd ${BTOR2TOOLS_DIR}
 case "$(uname -s)" in
    CYGWIN*|MINGW32*|MSYS*)
-     revision=8c150b3
-     echo ""
-     echo " *** WARNING: Building btor2tools on Windows relies on a specific"
-     echo " ***          revision (${revision})."
-     echo " ***          This version of Boolector may be built against an"
-     echo " ***          older version of btor2tools."
-     echo ""
-     git reset --hard "${revision}"
-     patch -p1 < ../../contrib/windows_patches/btor2tools_"${revision}".patch
+     component="btor2tools"
+     test_apply_patch "${component}" ../../contrib/windows_patches/${component}_20190110.patch
```

Comment:
Since the path `../../contrib/windows_patches/` is used in every setup script, you could define a `WINDOWS_PATCHES_DIR="$(pwd)/contribwindows_patches"` in `contrib/setup-utils.sh` and use `test_apply_patch "${component}" $WINDOWS_PATCHES_DIR/${component}_20190110.patch`

Reply:
Reasonable. I can do this.

I'm tempted to get rid of `test_apply_patch "${component}"` and then just work it based on:

```
local component=$(basename $(echo "../../contrib/windows_patches/${component}_20190110.patch") |  cut -d "_" -f 1)
```

would that be okay with you too? Right now, the double passing-in of `${component}` is superfluous.

Reply:
Actually, if you want to print the component name with the correct letter case, e.g., CaDiCaL, Lingeling etc. you probably have to pass something like:

```
test_apply_patch "Btor2Tools" "$WINDOWS_PATCHES_DIR/btor2tools_20190110.patch"
```

Reply:
Done!